tests/versioned/apollo-server-hapi/hapi-segments-tests.js

```diff
@@ -0,0 +1,621 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const { executeQuery, executeQueryBatch } = require('../test-client')
+
+const ANON_PLACEHOLDER = '<anonymous>'
+const UNKNOWN_OPERATION_PLACEHOLDER = '<operation unknown>'
+
+/**
+ * Creates a set of standard transction tests to run against various
+ * apollo-server libraries.
+ * It is required that t.context.helper and t.context.serverUrl are set.
+ * @param {*} t a tap test instance
+ */
+function createSegmentsTests(t) {
+  t.test('anonymous query, single level', (t) => {
+    const { helper, serverUrl } = t.context
+
+    const query = `query {
+      hello
+    }`
+
+    helper.agent.on('transactionFinished', (transaction) => {
+      const operationPart = `query ${ANON_PLACEHOLDER}`
+      const expectedSegments = [{
+        name: `WebTransaction/apollo-server/${operationPart} hello`,
+        children: [{
+          name: 'Nodejs/Middleware/Hapi/handler//gql',
```

Comment:
interesting the `//gql` is in there. prob related to where with express i saw adding an async function to the middle of resolver execution resulted in `//graphql` on a pile of middleware.